tests/test_core.py

```diff
@@ -262,3 +262,23 @@ def test_complex_struct(self):
         self.db['test'] = test_value
         assert self.db['test'] == test_value
         assert self.get_json('test') == json.dumps(test_value)
+
+
+class TablenamesTest(TestCaseBackport):
+
+    def test_tablenames(self):
+        fname = norm_file('tests/db/tablenames-test-1.sqlite')
+        db = sqlitedict.SqliteDict(fname) 
+        self.assertEqual(db.get_tablenames(), ['unnamed'])
+        db.close()
+
+        fname = norm_file('tests/db/tablenames-test-2.sqlite')
+        db1 = sqlitedict.SqliteDict(fname,tablename='table1')         
+        self.assertEqual(db1.get_tablenames(), ['table1'])
+        db2 = sqlitedict.SqliteDict(fname,tablename='table2') 
+        self.assertEqual(db1.get_tablenames(), ['table1','table2'])
+        db1.close()
```

Comment:
```
with SliteDict(...): 
    self.assertEqual 
```
better than db = ; db.close()

Reply:
will do that

Reply:
is there a reason you're not having 
`from sqlitedict import SliteDict`  ?

Reply:
No

Reply:
ok, I did, makes things look cleaner.
having `with` makes things look better indeed. However my test is now the only one doing that. looks a bit inconsistent...

Reply:
@transfluxus I think we'll cleanup codebase later, don't worry :)